File: e-proc-repository/src/main/java/id/co/iconpln/eproc/db/domain/xml/master/Bidang.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package id.co.iconpln.eproc.db.domain.xml.master;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

/**
 *
 * @author Stefu
 */

@XmlRootElement( name = "ROW" )
public class Bidang {
    
    private String kodeBidang;
    private String nama;
    private String keterangan;
    private Integer flagTampil;
    private String createDate;
    private String createBy;

    /**
     * @return the kodeBidang
     */
    public String getKodeBidang() {
        return kodeBidang;
    }

    /**
     * @param kodeBidang the kodeBidang to set
     */
    @XmlElement( name = "ID_BIDANG")
    public void setKodeBidang(String kodeBidang) {
        this.kodeBidang = kodeBidang;
    }

    /**
     * @return the nama
     */
    public String getNama() {
        return nama;
    }

    /**
     * @param nama the nama to set
     */
    @XmlElement( name = "NAMA")
    public void setNama(String nama) {
        this.nama = nama;
    }

    /**
     * @return the keterangan
     */
    public String getKeterangan() {
        return keterangan;
    }

    /**
     * @param keterangan the keterangan to set
     */
    @XmlElement( name = "KETERANGAN")
    public void setKeterangan(String keterangan) {
        this.keterangan = keterangan;
    }

    /**
     * @return the flagTampil
     */
    public Integer getFlagTampil() {
        return flagTampil;
    }

    /**
     * @param flagTampil the flagTampil to set
     */
    @XmlElement( name = "FLAG_TAMPIL")
    public void setFlagTampil(Integer flagTampil) {
        this.flagTampil = flagTampil;
    }

    /**
     * @return the createDate
     */
    public String getCreateDate() {
        return createDate;
    }

    /**
     * @param createDate the createDate to set
     */
    @XmlElement( name = "CREATE_DATE")
    public void setCreateDate(String createDate) {
        this.createDate = createDate;
    }

    /**
     * @return the createBy
     */
    public String getCreateBy() {
        return createBy;
    }

    /**
     * @param createBy the createBy to set
     */
    @XmlElement( name = "CREATE_BY")
    public void setCreateBy(String createBy) {
        this.createBy = createBy;
    }
    
    
}

File: e-proc-web/src/main/webapp/assets/admin/pages/master/aplikasi/bidang.js

/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

var selectedRow;
var table;

$(document).ready(
        function () {
            $("#master_bidang_list_btnadd").click(function () {
                $("#master_bidang_form_add").show("slow");
                $("#master_bidang_list").hide();
                $("#master_bidang_form_add form").trigger("reset");
                $("#master_bidang_form_add input[name='mode']").val("simpan");
            });

            $(".master_bidang_form_add_btncancel").click(function (e) {
                e.preventDefault();
                $("#master_bidang_list").show("slow");
                $("#master_bidang_form_add").hide();
                $("#master_bidang_form_edit").hide();
                table.ajax.reload();
            });

            $(".master_bidang_list_btnedit").click(function (e) {
                e.preventDefault();
                $("#master_bidang_form_add").show("slow");
                $("#master_bidang_list").hide();
                $("#master_bidang_form_add input[name='mode']").val("update");
            });
            
            hidePanelMessage();
            initFormKodeBidang();
            initDataTableBidang();
            //initDeleteBidang();
        });

function hidePanelMessage() {
    $('.panel-message').hide();
}

function initFormKodeBidang() {

    jQuery("#master_bidang_form_add form").validationEngine({
        custom_error_messages: {
            '#inKodeBidang': {
                'required': {
                    'message': "Kode Bidang Harus Diisi"
                },
                'custom[onlyLetterNumber]': {
                    'message': "Kode Bidang Hanya boleh diisi Huruf dan angka"
                }
            },
            '#inNamaBidang': {
                'required': {
                    'message': "Nama Bidang Harus Diisi"
                },
                'minSize': {
                    'message': "Nama Bidang minimal 5 karakter"
                }
            }
        },
        onValidationComplete: function (form, status) {
            if (status === true) {

                $.ajax({
                    type: "POST",
                    url: BASE_URL + "api/master/aplikasi/bidang/simpan",
                    data: form.serializeArray(),
                    success: function (response) {
                        console.log(response);
                        showList(
                                $('#master_bidang_form_add'),
                                response.SAVEREFBIDANG,
                                response.OUTMESSAGE
                                );
                        if (response.SAVEREFBIDANG === '1') {

                        } else {
                            //do nothing
                        }
                    }
                });
            } else {
                //alert('error');
            }
        }
    });
}

function initDataTableBidang(){

    table = $('#tb_master_bidang').DataTable( {
        "processing": true,
        "serverSide": true,
        "aoColumnDefs": [
            {
                "bSortable": false,
                "aTargets": [ 0,1,2,3,4]
            },
            {
                "aTargets": [ 4 ], // Column to target
                "mRender": function ( data, type, full ) {
                    var ret_value =
                        '<a href="#" class="widget-icon widget-icon-dark master_bidang_list_btnedit"><span class="icon-pencil"></span></a>' +
                        '<a href="#master_bidang_modal_delete" data-toggle="modal"  class="widget-icon widget-icon-dark"><span class="icon-trash"></span></a>'
                    return ret_value;
                }
            }
        ],
        "ajax": {
            "url": BASE_URL + "api/master/aplikasi/bidang/json",
            "type": "GET",
            "dataSrc" : function(json){
                return json.data.bidangList;
            }
        },
        "columns": [
            {"data": "ROW_NUMBER", "defaultContent": ""},
            {"data": "kodeBidang", "defaultContent": ""},
            {"data": "nama", "defaultContent": ""},
            {"data": "keterangan", "defaultContent": ""},
            {"data": "flagTampil", "defaultContent": ""},
            {"data": "kodeBidang", "defaultContent": ""}
        ],

        "drawCallback": function( settings ) {
            initDatatableAction();
        }
    } );

}

function initDatatableAction(){

    $(".master_bidang_list_btnedit").click(function(e) {
        e.preventDefault();
        $("#master_bidang_form_add").show("slow");
        $("#master_bidang_list").hide();
        $("#master_bidang_form_add input[name='mode']").val("update");
    });

    //var table = $('#tb_master_bank').DataTable();
    $('#tb_master_bidang tbody').on( 'click', 'tr', function () {
        selectedRow = table.row(this).data();
        console.log(selectedRow);
        fillForm(selectedRow);
    } );

    //252
    //88

    //telkom cab bandung, intan.//13.00, reza.fitrianto
}

function fillForm(selectedRow){

    console.log('selected row', selectedRow);
    $("#master_bidang_form_add input[name='inKodeBidang']").val(selectedRow.kodeBidang);
    $("#master_bidang_form_add input[name='inNamaBidang']").val(selectedRow.nama);
    $("#master_bidang_form_add input[name='inKeterangan']").val(selectedRow.keterangan);
    $("#master_bidang_form_add input[name='inFlagTampil']").val(selectedRow.alamat);

    $.ajax({
        type: "POST",
        url: BASE_URL+"api/master/aplikasi/bidang/get_by_kodebidang",
        data: {
            kodeBidang : selectedRow.kodeBidang
        },
        success: function (response) {
            console.log(response);
        }
    });

}

function showList(form, success, message){
    $(form).hide();
    $("#master_bidang_list").show('slow');

    //show message
    if(success==='1'){
        $('#message-success').show('slow');
        $('#message-error').hide();
    }else{
        $('#message-success').hide();
        $('#message-error').show('slow');
        $('#message-error .panel-footer').html(message);
    }
    table.ajax.reload();
}
